From 1e7b4b040913425501d9a00086e104fe1be0df2e Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Sun, 4 Apr 2010 06:22:43 -0500
Subject: [PATCH 01/21] v4l2sink: Improve behavior for shared buffers.

When the decoder is using pad_alloc(), v4l2sink would behave badly if
the number of buffers ('queue-size' property) was not high enough to
account for all the buffers needed by the decoder, and other elements
(such as queues) between the decoder and v4l2sink.  This patch
slightly increases the default number of buffers, and changes v4l2sink
to drop frames rather than return an error in case the number of
buffers is not high enough.
---
 sys/v4l2/gstv4l2sink.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/sys/v4l2/gstv4l2sink.c b/sys/v4l2/gstv4l2sink.c
index 679f7e2..b6f2e2f 100644
--- a/sys/v4l2/gstv4l2sink.c
+++ b/sys/v4l2/gstv4l2sink.c
@@ -56,7 +56,7 @@
 GST_DEBUG_CATEGORY (v4l2sink_debug);
 #define GST_CAT_DEFAULT v4l2sink_debug
 
-#define PROP_DEF_QUEUE_SIZE         8
+#define PROP_DEF_QUEUE_SIZE         12
 #define DEFAULT_PROP_DEVICE   "/dev/video1"
 
 enum
@@ -668,7 +668,9 @@ gst_v4l2sink_show_frame (GstBaseSink * bsink, GstBuffer * buf)
         &newbuf);
 
     if (GST_FLOW_OK != ret) {
-      return ret;
+      GST_DEBUG_OBJECT (v4l2sink,
+          "dropping frame!  Consider increasing 'queue-size' property!");
+      return GST_FLOW_OK;
     }
 
     memcpy (GST_BUFFER_DATA (newbuf),
-- 
1.7.1

